File: modified_scripts/r/binary_search/qwen3_8b.r
binary_search <- function(vec, target) {
  left <- 1
  right <- length(vec)
  while (left <= right) {
    mid <- (left + right) %/% 2
    if (vec[mid] == target) {
      return(mid)
    } else if (vec[mid] < target) {
      left <- mid + 1
    } else {
      right <- mid - 1
    }
  }
  return(-1)
}

File: modified_scripts/rust/binary_search/qwen3_8b.rs
fn binary_search(arr: &[i32], target: i32) -> Option<usize> {
    let mut left = 0;
    let mut right = arr.len() as i32 - 1;
    while left <= right {
        let mid = (left + right) / 2;
        if arr[mid as usize] == target {
            return Some(mid as usize);
        } else if arr[mid as usize] < target {
            left = mid + 1;
        } else {
            right = mid - 1;
        }
    }
    None
}